dotnet-desktop-guide/winforms/controls/how-to-make-thread-safe-calls.md

```diff
@@ -61,3 +71,62 @@ The example counts from 0 to 10 in the `DoWork` event, pausing for one second be
 
 :::code language="csharp" source="snippets/how-to-make-thread-safe-calls/cs/FormBackgroundWorker.cs" id="Background":::
 :::code language="vb" source="snippets/how-to-make-thread-safe-calls/vb/FormBackgroundWorker.vb" id="Background":::
+
+## Example: Use Control.InvokeAsync (.NET 9 and later)
+
+Starting with .NET 9, Windows Forms includes the <xref:System.Windows.Forms.Control.InvokeAsync%2A> method, which provides async-friendly marshaling to the UI thread. This method is particularly useful for async event handlers and eliminates many common deadlock scenarios.
+
+> [!NOTE]
+> `Control.InvokeAsync` is only available in .NET 9 and later. It is not supported in .NET Framework.
+
+### Understanding the difference: Invoke vs InvokeAsync
+
+**Control.Invoke (Sending - Blocking):**
+
+- Synchronously sends the delegate to the UI thread's message queue.
+- The calling thread waits until the UI thread processes the delegate.
+- Can lead to UI freezes if overused during long-running operations.
+- Useful when immediate results are needed from the UI thread.
+
+**Control.InvokeAsync (Posting - Non-blocking):**
+
+- Asynchronously posts the delegate to the UI thread's message queue.
+- The calling thread doesn't wait and continues its work immediately.
```

Comment:
```suggestion
- Rather than waiting for the Invoke to return in a blocking way, the calling thread schedules just a call-back, but returns immediately. Although the code flow seems to be waiting at the `await` - the calling thread remains responsive at all times, and the code flow continues automatically through that internally scheduled callback mechanism.
```